src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  // write your code her
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(bitten) {
+    if (!bitten.hidden && bitten instanceof Herbivore) {
+      bitten.health -= 50;
+
+      if (!bitten.health) {
+        Animal.alive = Animal.alive.filter((herbivore) => herbivore.health > 0);
+      }
```

Comment:
Lets move this to Animal class and put inside new method e.g. 'killAnimal' and then reuse it here 
```suggestion
    animal.killAnimal();
```
also, avoid nesting - putting if inside another if.